pkg/prometheus_ingester/prometheus_ingester_test.go

```diff
@@ -718,3 +720,122 @@ func Test_processHistogramIncrease(t *testing.T) {
 		assert.ElementsMatchf(t, testCase.expectedEvents, generatedEvents, "expected events:\n%s\n\nresult:\n%s", testCase.expectedEvents, generatedEvents)
 	}
 }
+
+func TestPrometheusIngesterHttpHeader_UnmarshalYAML(t *testing.T) {
+	const envName = "ENVNAME"
+	var expectedValue = "value"
+	var expectedValueWithPrefix = "xxxvalue"
+
+	if err := os.Setenv(envName, expectedValue); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name          string
+		data          []byte
+		expectedValue *string
+		wantErr       bool
+	}{
+		{
+			name: "value from string",
+			data: []byte(`
+name: name
+value: value
+`),
+			expectedValue: &expectedValue,
+		},
+		{
+			name: "value from env",
+			data: []byte(`
+name: name
+valueFromEnv:
+  name: ENVNAME
+`),
+			expectedValue: &expectedValue,
+		},
+		{
+			name: "value from env with valuePrefix",
+			data: []byte(`
+name: name
+valueFromEnv:
+  name: ENVNAME
+  valuePrefix: xxx
+`),
+			expectedValue: &expectedValueWithPrefix,
+		},
+		{
+			name: "both valueFromString and valueFromEnv is set",
+			data: []byte(`
+name: name
+valueFromEnv: 
+  name: ENVNAME
+value: value
+`),
+			wantErr: true,
+		},
+		{
+			name:    "none of valueFromString or valueFromEnv is set",
+			data:    []byte("name: name"),
+			wantErr: true,
+		},
+		{
+			name:    "none of valueFromString or valueFromEnv is set",
+			data:    []byte("value: value"),
+			wantErr: true,
+		},
+		{
+			name: "non existing environment variable with name valueFromEnv",
+			data: []byte(`
+name: name
+valueFromEnv:
+  name: NON_EXISTING_ENV_NAME
```

Comment:
Tests should not depend on the environment they are run in.

This variable could exist, the test should explicitly unset it.

Reply:
Not fixed properly :)

Reply:
Fixed in b1a9e225fa8a9c1e31ebeeada50917c0bd2404f9.